### DISCUSSION TITLE
consult-register not working anymore

### DISCUSSION THREAD

**Discussion by MarcRohrer** [minad/consult#941] [Q&A]
Hi minad,

since the last update consult-register does not work for me anymore.

I get:

```
Autoloading file c:/emacs/.emacs.d/elpa/consult-20240209.2104/consult.elc failed to define function consult-register
```
I reinstalled consult and also recompiled consult.el as well as consult-register.el, but no change.
Am I doing something wrong?

My config:

```
 (use-package consult
        :bind (("M-i" . consult-ripgrep)
               ("<f4>" . consult-register)
               ("C-g" . consult-goto-line)
               ("<" . consult-narrow-key)
               (">" . consult-widen-key))
        :hook (completion-list-mode . consult-preview-at-point-mode)
        :init
        (setq consult-narrow-key "<")
        (setq consult-widen-key ">")
        (setq xref-show-xrefs-function #'consult-xref
              xref-show-definitions-function #'consult-xref))
```
Thank you very much for your time!

Marc

**Comment by minad** [comment]
It seems the problem is that `consult-register` is defined in consult-register.el but for some reason your setup assumes that it should be in consult.el. This problem should not occur since `consult-register` is autoloaded. Maybe your autoloads are not correctly generated after package installation?

**Reply by MarcRohrer** [reply]
Hi,

I have absolutely no clue, why or how this could happen 
or how I could diagnose this... :-(

Marc

**Reply by minad** [reply]
In case it is not possible to diagnose the problem otherwise I recommend to reinstall the affected packages. See also https://github.com/minad/consult?tab=readme-ov-file#bug-reports.

**Reply by MarcRohrer** [reply]
Hi Minad,

 I am a devoted user of a lot of your packages, which i find HIGHLY useful and perfectly crafted!
I understand, that this is annoys for you.

I reinstalled Emacs, all packages and the specific packages multiple times and also recompiled consult.el and consult-register.el manually.

It might be possible to diagnose the problem further, only I have absolutely no clue, how to do it. Why shouldn't the autoload be working? Wasn't that way until a few days ago. Now something changed, but I do not know what :-(

consult-imenu is also not loaded...
while consult-buffer works fine.

platform: windows
GNU Emacs 29.1 (build 2, x86_64-w64-mingw32)
(completion-styles '(orderless basic))

I have followed your recommendations with the minimal setup. After emacs -Q and

```
(package-initialize)
nil
(require 'consult)
consult
(require 'vertico)
vertico
(vertico-mode)
t
(setq completion-styles '(substring basic))
(substring basic)
(consult-register)
```

I still get:

```
Debugger entered--Lisp error: (void-function consult-register)
  (consult-register)
  (progn (consult-register))
  elisp--eval-last-sexp(t)
  eval-last-sexp(t)
  eval-print-last-sexp(nil)
  funcall-interactively(eval-print-last-sexp nil)
  command-execute(eval-print-last-sexp)
```

sorry for the inconvenience!

Marc

**Reply by minad** [reply]
The problem is that your autoloads don't work. I don't know either why that happens. You can try to `(require 'consult-autoloads)`. If this fails, the autoloads were not generated properly in the first place, if it works after that the question remains why the autoloads are not loaded automatically as they should.

**Reply by MarcRohrer** [reply]
Hi,

so it looks like. With ```(require 'consult-autoloads)``` everything works fine immediately. I added that to my .emacs and it works as usual now. Yes, that question remains. Very, very strange :-(

Thanx so much for your help!

Marc

**Reply by oantolin** [reply]
Oh, did you recently change your configuration so `package-initialize` is no longer called during start up, @MarcRohrer? Sometimes people do that for "efficiency" but I don't think it's worth the trouble, because then all autoloads become your responsibility.

**Comment by MarcRohrer** [comment]
Hi,not really 😳Thanx!MarcAm 15.02.2024 19:10 schrieb Omar Antolín Camarena ***@***.***>:
Oh, did you recently change your configuration so package-initialize is no longer called during start up, @MarcRohrer. Sometimes people do that for "efficiency" but I don't think it's worth the trouble, because then all autoloads become your responsibility.
